design/milestone_1_2_design_doc.md

```diff
@@ -0,0 +1,455 @@
+| 2         | *Vanilla flow*, non-conflicting Secrets Provider<br />       | Two Secrets Providers have access to different Conjur secret | 2 Secrets Provider Jobs run in same namespace | - All relevant K8s secrets are updated<br />- Verify logs    |
+| 3         | Non-conflicting Secrets Provider 1 namespace same K8s Secret * | Two Secrets Providers have access to same secret             | 2 Secrets Provider Jobs run in same namespace | - No race condition and Secrets Providers will not override each other<br />- Verify logs |
+| 4         | *Regression tests*                                           | All regression tests should pass                             |                                               |                                                              |
+
+### Performance tests
+
+| Title | Given                     | When                                                         | Then                           | Comment                                                      |
+| ----- | ------------------------- | ------------------------------------------------------------ | ------------------------------ | ------------------------------------------------------------ |
+| 1     | Performance investigation | - 1000 Secrets Providers defined<br />- Conjur secrets with max amount of characters | Secrets Provider runs as a Job | Evaluate how many K8s Secrets can be updated with Conjur secrets in 5 minutes |
+| 2     | Performance investigation | - 1000 Secrets Providers defined<br />- Conjur secret with average amount of characters | Secrets Provider runs as a Job | Evaluate how many K8s Secrets can be updated with Conjur secrets in 5 minutes |
+
+
+
+## Logs
+
+| **Scenario**                         | **Log message**                                              |
+| ------------------------------------ | ------------------------------------------------------------ |
+| Secrets Provider spins up            | Kubernetes Secrets Provider v\*%s\* starting up as a %s... (Job/Deployment) |
+| Job has completed and is terminating | Kubernetes Secrets Provider has finished task and is terminating… |
+
+### Audit 
+
+All fetches on a Conjur Resource are individually audited, creating its own audit entry. Therefore there will be no changes in audit behavior for this Milestone.
+
+## Documentation
+
+
+
+## Open questions
+
+1. Once the Job terminates, there is no way to access the Job's logs. **ttlSecondsAfterFinished** allows the Job to stay alive even after the task terminates. Do we want to supply a **ttlSecondsAfterFinished** default for customers so that they can evaluate Job logs even after the Job terminates?
```

Comment:
this is quite a big question. will this happen also if the secrets-provider failed to retrieve secrets or is it just after a successful cycle?

Also - i've seen that the default value you gave is `0`. Does that mean that by default the user will not be able to see logs? Is this the behaviour we want?

Reply:
If the secrets provider fails, it will behave like how a normal pod that fails behaves. `ttlSecondsAfterFinished` is for finished resources

> Does that mean that by default the user will not be able to see logs? 
Yes

> Is this the behaviour we want?
No not a good idea. How long do you think it should stay up? How long does our init container stay up before terminating? 60 secs?

@oburstein-hub what do you think the default `ttlSecondsAfterFinished` should be?

Reply:
@InbalZilberman @oburstein-hub 
I notice [here](https://kubernetes.io/docs/concepts/workloads/controllers/job/#ttl-mechanism-for-finished-jobs) that `ttlSecondsAfterFinished` is an alpha feature.
I see here that in order to use this feature the customer might need to enable alpha features in their GKE cluster with something like:
`kube-apiserver --feature-gates=TTLAfterFinished=true` AND in their kube-controller-manager because it isn't enabled by default

Reply:
If it is alpha feature we cannot use it at all IMO
I will look to find common ways of retrieving logs for Job

Reply:
Please see what I found: When a Job completes, no more Pods are created, but the Pods are not deleted either. Since they are terminated, they don’t show up with kubectl get pods, but they will show up with kubectl get pods -a. Keeping them around allows you to still view the logs of completed pods

Reply:
Sigal - please try this, and if it works you don't need any special parameter to see logs of job that finished to run

Reply:
Nice @oburstein-hub ! then in this case we just need to update our docs to mention that.

Reply:
yeah, nice find! also came across this but I will double check this for sanity purposes